services/agents/agents.go

```diff
@@ -40,3 +43,32 @@ func redactWords(agent *models.Agent) []string {
 	}
 	return words
 }
+
+// filterOutCollectors removes from exporter's flags disabled collectors.
+// DisableCollector values should  match collector flag till end of string or till `=` sign.
+// Examples:
+// 1. if we pass `meminfo` then only "--collector.meminfo" but not "--collector.meminfo_numa"
+// 2. if we pass `netstat.field` then "--collector.netstat.fields=^(.*_(InErrors|InErrs|InCsumErrors)..." should be disabled.
+// 3. To disable "--collect.custom_query.hr" with directory ""--collect.custom_query.lr.directory" user should pass both names.
+func filterOutCollectors(prefix string, args, disabledCollectors []string) []string {
+	argsMap := make(map[string]string)
+	for _, arg := range args {
+		flagName := strings.Split(arg, "=")[0]
+		argsMap[flagName] = arg
+	}
+
+	for _, disabledCollector := range disabledCollectors {
+		key := fmt.Sprintf("%s%s", prefix, disabledCollector)
+		_, ok := argsMap[key]
+		if ok {
+			delete(argsMap, key)
+		}
+	}
+
+	enabledArgs := []string{}
+	for _, arg := range argsMap {
+		enabledArgs = append(enabledArgs, arg)
+	}
+
+	return enabledArgs
```

Comment:
```suggestion
	disabledCollectorsMap := make(map[string]string)
	for _, disabledCollector := range disabledCollectors {
		key := fmt.Sprintf("%s%s", prefix, disabledCollector)
		disabledCollectorsMap[key] = struct{}
	}

	enabledArgs := []string{}
	for _, arg := range args {
		flagName := strings.Split(arg, "=")[0]
		if _, ok := disabledCollectorsMap[flagName]; !ok {
			enabledArgs = append(enabledArgs, arg)
		}
	}

	return enabledArgs
```

Reply:
The same in the other place. But I'm almost sure that we can extract this to utils and call it from both places.